app/actions/local/user.ts

```diff
@@ -105,25 +105,39 @@ export const updateRecentCustomStatuses = async (serverUrl: string, customStatus
     });
 };
 
-export const updateUserPresence = async (serverUrl: string, userStatus: UserStatus) => {
-    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
-    if (!operator) {
+export const updateLocalUser = async (serverUrl: string, userId: string, userDetails: Partial<UserProfile> & { status?: string}) => {
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
         return {error: `${serverUrl} database not found`};
     }
 
-    const user = await queryUserById(operator.database, userStatus.user_id);
-    if (user) {
-        user.prepareUpdate((record) => {
-            record.status = userStatus.status;
+    try {
+        const user = await database.get(MM_TABLES.SERVER.USER).find(userId) as UserModel;
+        await database.write(async () => {
+            await user.update((userRecord: UserModel) => {
+                userRecord.status = userDetails?.status ?? user.status;
+                userRecord.authService = userDetails.auth_service ?? user.authService;
+                userRecord.deleteAt = userDetails.delete_at ?? user.deleteAt;
```

Comment:
you are not really deleting the user locally like this unless the server tells you to delete it, basically this will never happen here